### ISSUE TITLE
Crypto provider for Rustls

### ISSUE THREAD

**Issue by howardjohn** [cloudflare/boring#165]
Recently, Rustls was updated to have allow a pluggable crypto provider: https://github.com/rustls/rustls/pull/1405/files.

I believe it would now be possible to plugin `boring` into this. This is appealing as we get the benefits of boringssl as well as rustls - notably, FIPS compliance. 

Is there interest in providing such a provider?

**Comment by janrueth** [comment]
I tried to hack something together over the weekend that uses boring as the crypto provider: https://github.com/janrueth/boring-rustls-provider

**Comment by rushilmehra** [comment]
Yeah I think this is a good idea. @janrueth thanks for hacking that together, I'll give it a look and we can go from there

**Comment by Sytten** [comment]
Any planned progress on bringing that provider crate in this repo?
